test/unit/fact_importer_test.rb

```diff
@@ -55,6 +55,73 @@ def fact_name_class
     end
   end
 
+  describe '#normalize' do
+    test 'filters regular facts' do
+      Setting.stubs(:[]).with(:excluded_facts).returns(['ignore*', '*_bad', 'filter'])
+
+      facts = {
+        'ignored_fact' => 'will_not_show',
+        'fact_bad' => 'will_not_show',
+        'filter' => 'will_not_show',
+        'just_a_fact' => 'hello',
+      }
+
+      importer = FactImporter.new(nil, facts)
+      actual_facts = importer.send(:facts)
+
+      assert_not_nil actual_facts['just_a_fact']
+      assert_equal 1, actual_facts.count
+    end
+
+    test 'filters a::b facts' do
+      Setting.stubs(:[]).with(:excluded_facts).returns(['ignore*', '*_bad', 'filter'])
+
+      facts = {
+        # rhsm style flat facts: a::b::c
+        'ignored_fact::something' => 'will_not_show',
+        'something::ignored_fact' => 'will_not_show',
+        'something::ignored_fact::something_else' => 'will_not_show',
+        'fact_bad::something' => 'will_not_show',
+        'something::fact_bad' => 'will_not_show',
+        'something::fact_bad::something_else' => 'will_not_show',
+        'filter::something' => 'will_not_show',
+        'something::filter' => 'will_not_show',
+        'something::filter::something_else' => 'will_not_show',
+        'something::something_else' => 'hello'
+      }
+
+      importer = FactImporter.new(nil, facts)
+      actual_facts = importer.send(:facts)
+
+      assert_not_nil actual_facts['something::something_else']
+      assert_equal 1, actual_facts.count
+    end
+
+    test 'filters macaddress_virbr0 facts' do
+      Setting.stubs(:[]).with(:excluded_facts).returns(['ignore*', '*_bad', 'filter'])
+
+      facts = {
+        # facter < 3.0 style facts: macaddress_virbr0
+        'ignored_fact_something' => 'will_not_show',
+        'something_ignored_fact' => 'will_not_show',
+        'something_ignored_fact_something_else' => 'will_not_show',
+        'something_fact_bad' => 'will_not_show',
+        'something_filter' => 'will_not_show',
+        'something_filter_something_else' => 'hello',
+        'fact_bad_something' => 'hello',
+        'filter_something' => 'hello'
+      }
```

Comment:
Also edge cases: "ignore" and "_bad" and "filter" names should all be filtered. Although it might not make sense, users could try to filter some facts with this.